### ISSUE TITLE
pkg stalls on exit

### ISSUE THREAD

**Issue by AdSchellevis** [opnsense/core#4397]
**Important notices**
Before you add a new report, we ask you kindly to acknowledge the following:

[x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md

[x] I have searched the existing issues and I'm convinced that mine is new.

**Describe the bug**
Although core might not be the best place to open this report, I thought to add it here so we don't forget to look at this (preferably before a next release).

Since pkg 1.15.6 a package install sometimes freezes on exit. At first I thought it was related to syslog-ng spawning it's own workers which pkg didn't know about, but when disabling the lockout handler the issue seems to remain.

**To Reproduce**

Easiest option to reproduce is to go to a small plugin and `make upgrade` a couple of times, after a couple of times it always seems to freeze. Currently I don't suspect syslog-ng of not wanting to restart, because it hangs after configuring system logging.

```
Configuring system logging...done.
```

When running the command in truss (`truss make upgrade`), it doesn't seem to be stalling on my end, which makes debugging a bit difficult. When looking at the hanging process with truss, it shows the following:

```
# fstat -p 42816 -v
USER     CMD          PID   FD MOUNT      INUM MODE         SZ|DV R/W
root     pkg        42816 text /        7065390 -rwxr-xr-x  3069048  r
root     pkg        42816 ctty /dev        132 crw--w----   pts/1 rw
root     pkg        42816   wd /        7713449 drwxr-xr-x     512  r
root     pkg        42816 root /             2 drwxr-xr-x    1024  r
root     pkg        42816    0 /dev        132 crw--w----   pts/1 rw
root     pkg        42816    1 /dev        132 crw--w----   pts/1 rw
root     pkg        42816    2 /dev        132 crw--w----   pts/1 rw
root     pkg        42816    3 /             2 drwxr-xr-x    1024  r
root     pkg        42816    4 /        2969888 drwxr-xr-x     512  r
root     pkg        42816    5 /        2969889 -rw-r--r--  27588608 rw
root     pkg        42816    6 /        7729592 -rw-r--r--    2888  r
root     pkg        42816    7 /             2 drwxr-xr-x    1024  r
root     pkg        42816    8* local seqpak fffff800043b2500 <-> fffff800a7578500
```

```
# truss -p 42816
read(8,"\0\0\0",8192)				 = 3 (0x3)
SIGNAL 18 (SIGTSTP) code=SI_KERNEL
read(8,0x3397b09e000,8192)			 = 0 (0x0)
close(8)					 = 0 (0x0)
access("/etc/localtime",R_OK)			 = 0 (0x0)
```

**Expected behavior**
It (pkg) should probably just exit, but if restarting services in pkg causes issues more frequently, we might look into alternatives (detach the restart process?). I'm not 100% sure how this is supposed to work, for some reason I haven't seen it crash when restarting configd for example.

**Screenshots**
-
**Relevant log files**
-
**Additional context**
-

**Environment**

OPNsense 20.7.3 (amd64, OpenSSL).

**Comment by mimugmail** [comment]
I did several updates today, one thing which looks weird is this:
```
Installed packages to be REINSTALLED:
	ntp-4.2.8p15 (direct dependency changed: perl5)
	pkg-1.15.6
	rrdtool-1.7.2_4 (direct dependency changed: perl5)
	rtrlib-0.6.3 (needed shared library changed)

Number of packages to be installed: 2
Number of packages to be upgraded: 52
Number of packages to be reinstalled: 4

The process will require 5 MiB more space.
4 MiB to be downloaded.
[1/1] Fetching pkg-1.15.6.txz: .......... done
[1/60] Upgrading libffi from 3.3 to 3.3_1...
[1/60] Extracting libffi-3.3_1: .......... done
[2/60] Upgrading python37 from 3.7.8_1 to 3.7.9...
[2/60] Extracting python37-3.7.9: .......... done
```

Fetching pkg but not installing ... 
This is the complete output:

```
***GOT REQUEST TO UPGRADE: all***
Updating OPNsense repository catalogue...
OPNsense repository is up to date.
All repositories are up to date.
Updating OPNsense repository catalogue...
OPNsense repository is up to date.
All repositories are up to date.
Checking for upgrades (57 candidates): .......... done
Processing candidates (57 candidates): .......... done
The following 60 package(s) will be affected (of 0 checked):

New packages to be INSTALLED:
	bash: 5.0.18_3
	bash-completion: 2.11,2
	zabbix5-agent: 5.0.3

Installed packages to be UPGRADED:
	ca_root_nss: 3.55 -> 3.57
	curl: 7.71.1 -> 7.72.0
	fping: 4.2 -> 5.0_1
	frr7: 7.4_1 -> 7.4_2
	gettext-runtime: 0.20.2 -> 0.21
	gnutls: 3.6.14 -> 3.6.15
	isc-dhcp44-relay: 4.4.2 -> 4.4.2_1
	isc-dhcp44-server: 4.4.2 -> 4.4.2_1
	json-c: 0.14 -> 0.15
	libffi: 3.3 -> 3.3_1
	libssh: 0.9.4 -> 0.9.5
	libunwind: 20200331 -> 20200331_1
	mpd5: 5.8_10 -> 5.9
	net-snmp: 5.7.3_20,1 -> 5.9,1
	nspr: 4.27 -> 4.29
	nss: 3.55 -> 3.57
	openldap-sasl-client: 2.4.50 -> 2.4.51
	opnsense: 20.7 -> 20.7.3
	opnsense-update: 20.7 -> 20.7.3
	os-dyndns: 1.22 -> 1.23
	os-frr: 1.14 -> 1.17
	os-zabbix-agent: 1.7 -> 1.8
	p11-kit: 0.23.20_1 -> 0.23.21
	perl5: 5.30.3 -> 5.32.0
	php73: 7.3.20 -> 7.3.22
	php73-ctype: 7.3.20 -> 7.3.22
	php73-curl: 7.3.20 -> 7.3.22
	php73-dom: 7.3.20 -> 7.3.22
	php73-filter: 7.3.20 -> 7.3.22
	php73-gettext: 7.3.20 -> 7.3.22
	php73-hash: 7.3.20 -> 7.3.22
	php73-json: 7.3.20 -> 7.3.22
	php73-ldap: 7.3.20 -> 7.3.22
	php73-openssl: 7.3.20 -> 7.3.22
	php73-pdo: 7.3.20 -> 7.3.22
	php73-session: 7.3.20 -> 7.3.22
	php73-simplexml: 7.3.20 -> 7.3.22
	php73-sockets: 7.3.20 -> 7.3.22
	php73-sqlite3: 7.3.20 -> 7.3.22
	php73-xml: 7.3.20 -> 7.3.22
	php73-zlib: 7.3.20 -> 7.3.22
	py37-cffi: 1.14.0_1 -> 1.14.3
	py37-six: 1.14.0 -> 1.15.0
	py37-sqlite3: 3.7.8_7 -> 3.7.9_7
	py37-urllib3: 1.25.7,1 -> 1.25.10,1
	python37: 3.7.8_1 -> 3.7.9
	rate: 0.9_1 -> 0.9_2
	sqlite3: 3.32.3_1,1 -> 3.33.0,1
	squid: 4.11_2 -> 4.13
	syslog-ng327: 3.27.1_1 -> 3.27.1_2
	trousers: 0.3.14_2 -> 0.3.14_3
	unbound: 1.10.1 -> 1.11.0
	zabbix4-agent: 4.0.22 -> 4.0.24
	zabbix4-proxy: 4.0.22 -> 4.0.24

Installed packages to be REINSTALLED:
	ntp-4.2.8p15 (direct dependency changed: perl5)
	rrdtool-1.7.2_4 (direct dependency changed: perl5)
	rtrlib-0.6.3 (needed shared library changed)

Number of packages to be installed: 3
Number of packages to be upgraded: 54
Number of packages to be reinstalled: 3

The process will require 6 MiB more space.
65 MiB to be downloaded.
[1/60] Fetching zabbix4-proxy-4.0.24.txz: .......... done
[2/60] Fetching zabbix4-agent-4.0.24.txz: .......... done
[3/60] Fetching unbound-1.11.0.txz: .......... done
[4/60] Fetching trousers-0.3.14_3.txz: .......... done
[5/60] Fetching syslog-ng327-3.27.1_2.txz: .......... done
[6/60] Fetching squid-4.13.txz: .......... done
[7/60] Fetching sqlite3-3.33.0,1.txz: .......... done
[8/60] Fetching rtrlib-0.6.3.txz: ...... done
[9/60] Fetching rrdtool-1.7.2_4.txz: .......... done
[10/60] Fetching rate-0.9_2.txz: ....... done
[11/60] Fetching python37-3.7.9.txz: .......... done
[12/60] Fetching py37-urllib3-1.25.10,1.txz: .......... done
[13/60] Fetching py37-sqlite3-3.7.9_7.txz: .... done
[14/60] Fetching py37-six-1.15.0.txz: ... done
[15/60] Fetching py37-cffi-1.14.3.txz: .......... done
[16/60] Fetching php73-zlib-7.3.22.txz: ... done
[17/60] Fetching php73-xml-7.3.22.txz: ... done
[18/60] Fetching php73-sqlite3-7.3.22.txz: ... done
[19/60] Fetching php73-sockets-7.3.22.txz: ..... done
[20/60] Fetching php73-simplexml-7.3.22.txz: ... done
[21/60] Fetching php73-session-7.3.22.txz: ..... done
[22/60] Fetching php73-pdo-7.3.22.txz: ...... done
[23/60] Fetching php73-openssl-7.3.22.txz: ........ done
[24/60] Fetching php73-ldap-7.3.22.txz: .... done
[25/60] Fetching php73-json-7.3.22.txz: ... done
[26/60] Fetching php73-hash-7.3.22.txz: .......... done
[27/60] Fetching php73-gettext-7.3.22.txz: . done
[28/60] Fetching php73-filter-7.3.22.txz: ... done
[29/60] Fetching php73-dom-7.3.22.txz: ........ done
[30/60] Fetching php73-curl-7.3.22.txz: .... done
[31/60] Fetching php73-ctype-7.3.22.txz: . done
[32/60] Fetching php73-7.3.22.txz: .......... done
[33/60] Fetching perl5-5.32.0.txz: .......... done
[34/60] Fetching p11-kit-0.23.21.txz: .......... done
[35/60] Fetching os-zabbix-agent-1.8.txz: .. done
[36/60] Fetching os-frr-1.17.txz: ....... done
[37/60] Fetching os-dyndns-1.23.txz: .... done
[38/60] Fetching opnsense-update-20.7.3.txz: ........ done
[39/60] Fetching opnsense-20.7.3.txz: .......... done
[40/60] Fetching openldap-sasl-client-2.4.51.txz: .......... done
[41/60] Fetching ntp-4.2.8p15.txz: .......... done
[42/60] Fetching nss-3.57.txz: .......... done
[43/60] Fetching nspr-4.29.txz: .......... done
[44/60] Fetching net-snmp-5.9,1.txz: .......... done
[45/60] Fetching mpd5-5.9.txz: .......... done
[46/60] Fetching libunwind-20200331_1.txz: .......... done
[47/60] Fetching libssh-0.9.5.txz: .......... done
[48/60] Fetching libffi-3.3_1.txz: ..... done
[49/60] Fetching json-c-0.15.txz: ......... done
[50/60] Fetching isc-dhcp44-server-4.4.2_1.txz: .......... done
[51/60] Fetching isc-dhcp44-relay-4.4.2_1.txz: .......... done
[52/60] Fetching gnutls-3.6.15.txz: .......... done
[53/60] Fetching gettext-runtime-0.21.txz: .......... done
[54/60] Fetching frr7-7.4_2.txz: .......... done
[55/60] Fetching fping-5.0_1.txz: .... done
[56/60] Fetching curl-7.72.0.txz: .......... done
[57/60] Fetching ca_root_nss-3.57.txz: .......... done
[58/60] Fetching bash-completion-2.11,2.txz: .......... done
[59/60] Fetching bash-5.0.18_3.txz: .......... done
[60/60] Fetching zabbix5-agent-5.0.3.txz: .......... done
Checking integrity... done (2 conflicting)
  - zabbix5-agent-5.0.3 conflicts with zabbix4-agent-4.0.24 on /usr/local/bin/zabbix_get
  - zabbix5-agent-5.0.3 conflicts with zabbix4-agent-4.0.22 on /usr/local/bin/zabbix_get
Cannot solve problem using SAT solver, trying another plan
Checking integrity... done (0 conflicting)
Conflicts with the existing packages have been found.
One more solver iteration is needed to resolve them.
The following 60 package(s) will be affected (of 0 checked):

New packages to be INSTALLED:
	bash: 5.0.18_3
	bash-completion: 2.11,2

Installed packages to be UPGRADED:
	ca_root_nss: 3.55 -> 3.57
	curl: 7.71.1 -> 7.72.0
	fping: 4.2 -> 5.0_1
	frr7: 7.4_1 -> 7.4_2
	gettext-runtime: 0.20.2 -> 0.21
	gnutls: 3.6.14 -> 3.6.15
	isc-dhcp44-relay: 4.4.2 -> 4.4.2_1
	isc-dhcp44-server: 4.4.2 -> 4.4.2_1
	json-c: 0.14 -> 0.15
	libffi: 3.3 -> 3.3_1
	libssh: 0.9.4 -> 0.9.5
	libunwind: 20200331 -> 20200331_1
	mpd5: 5.8_10 -> 5.9
	net-snmp: 5.7.3_20,1 -> 5.9,1
	nspr: 4.27 -> 4.29
	nss: 3.55 -> 3.57
	openldap-sasl-client: 2.4.50 -> 2.4.51
	opnsense: 20.7 -> 20.7.3
	opnsense-update: 20.7 -> 20.7.3
	os-dyndns: 1.22 -> 1.23
	os-frr: 1.14 -> 1.17
	p11-kit: 0.23.20_1 -> 0.23.21
	perl5: 5.30.3 -> 5.32.0
	php73: 7.3.20 -> 7.3.22
	php73-ctype: 7.3.20 -> 7.3.22
	php73-curl: 7.3.20 -> 7.3.22
	php73-dom: 7.3.20 -> 7.3.22
	php73-filter: 7.3.20 -> 7.3.22
	php73-gettext: 7.3.20 -> 7.3.22
	php73-hash: 7.3.20 -> 7.3.22
	php73-json: 7.3.20 -> 7.3.22
	php73-ldap: 7.3.20 -> 7.3.22
	php73-openssl: 7.3.20 -> 7.3.22
	php73-pdo: 7.3.20 -> 7.3.22
	php73-session: 7.3.20 -> 7.3.22
	php73-simplexml: 7.3.20 -> 7.3.22
	php73-sockets: 7.3.20 -> 7.3.22
	php73-sqlite3: 7.3.20 -> 7.3.22
	php73-xml: 7.3.20 -> 7.3.22
	php73-zlib: 7.3.20 -> 7.3.22
	py37-cffi: 1.14.0_1 -> 1.14.3
	py37-six: 1.14.0 -> 1.15.0
	py37-sqlite3: 3.7.8_7 -> 3.7.9_7
	py37-urllib3: 1.25.7,1 -> 1.25.10,1
	python37: 3.7.8_1 -> 3.7.9
	rate: 0.9_1 -> 0.9_2
	sqlite3: 3.32.3_1,1 -> 3.33.0,1
	squid: 4.11_2 -> 4.13
	syslog-ng327: 3.27.1_1 -> 3.27.1_2
	trousers: 0.3.14_2 -> 0.3.14_3
	unbound: 1.10.1 -> 1.11.0
	zabbix4-proxy: 4.0.22 -> 4.0.24

Installed packages to be REINSTALLED:
	ntp-4.2.8p15 (direct dependency changed: perl5)
	pkg-1.15.6
	rrdtool-1.7.2_4 (direct dependency changed: perl5)
	rtrlib-0.6.3 (needed shared library changed)

Number of packages to be installed: 2
Number of packages to be upgraded: 52
Number of packages to be reinstalled: 4

The process will require 5 MiB more space.
4 MiB to be downloaded.
[1/1] Fetching pkg-1.15.6.txz: .......... done
[1/60] Upgrading libffi from 3.3 to 3.3_1...
[1/60] Extracting libffi-3.3_1: .......... done
[2/60] Upgrading python37 from 3.7.8_1 to 3.7.9...
[2/60] Extracting python37-3.7.9: .......... done
[3/60] Upgrading gettext-runtime from 0.20.2 to 0.21...
[3/60] Extracting gettext-runtime-0.21: .......... done
[4/60] Installing bash-5.0.18_3...
[4/60] Extracting bash-5.0.18_3: .......... done
[5/60] Upgrading ca_root_nss from 3.55 to 3.57...
[5/60] Extracting ca_root_nss-3.57: ...... done
[6/60] Installing bash-completion-2.11,2...
[6/60] Extracting bash-completion-2.11,2: .......... done
[7/60] Upgrading trousers from 0.3.14_2 to 0.3.14_3...
===> Creating groups.
Using existing group '_tss'.
===> Creating users
Using existing user '_tss'.
[7/60] Extracting trousers-0.3.14_3: .......... done
[8/60] Upgrading php73 from 7.3.20 to 7.3.22...
[8/60] Extracting php73-7.3.22: .......... done
[9/60] Upgrading perl5 from 5.30.3 to 5.32.0...
[9/60] Extracting perl5-5.32.0: .......... done
[10/60] Upgrading p11-kit from 0.23.20_1 to 0.23.21...
[10/60] Extracting p11-kit-0.23.21: .......... done
[11/60] Upgrading libssh from 0.9.4 to 0.9.5...
[11/60] Extracting libssh-0.9.5: .......... done
[12/60] Upgrading sqlite3 from 3.32.3_1,1 to 3.33.0,1...
[12/60] Extracting sqlite3-3.33.0,1: .......... done
[13/60] Upgrading nspr from 4.27 to 4.29...
[13/60] Extracting nspr-4.29: .......... done
[14/60] Reinstalling rtrlib-0.6.3...
[14/60] Extracting rtrlib-0.6.3: .......... done
[15/60] Upgrading php73-pdo from 7.3.20 to 7.3.22...
[15/60] Extracting php73-pdo-7.3.22: .......... done
[16/60] Upgrading php73-json from 7.3.20 to 7.3.22...
[16/60] Extracting php73-json-7.3.22: .......... done
[17/60] Upgrading php73-hash from 7.3.20 to 7.3.22...
[17/60] Extracting php73-hash-7.3.22: .......... done
[18/60] Upgrading openldap-sasl-client from 2.4.50 to 2.4.51...
[18/60] Extracting openldap-sasl-client-2.4.51: .......... done
[19/60] Upgrading net-snmp from 5.7.3_20,1 to 5.9,1...
[19/60] Extracting net-snmp-5.9,1: .......... done
[20/60] Upgrading json-c from 0.14 to 0.15...
[20/60] Extracting json-c-0.15: .......... done
[21/60] Upgrading gnutls from 3.6.14 to 3.6.15...
[21/60] Extracting gnutls-3.6.15: .......... done
[22/60] Upgrading curl from 7.71.1 to 7.72.0...
[22/60] Extracting curl-7.72.0: .......... done
[23/60] Reinstalling pkg-1.15.6...
[23/60] Extracting pkg-1.15.6: .......... done
[23/60] Installing zabbix4-agent-4.0.24...
===> Creating groups.
Using existing group 'zabbix'.
===> Creating users
Using existing user 'zabbix'.
[23/60] Extracting zabbix4-agent-4.0.24: ......... done
[24/60] Upgrading unbound from 1.10.1 to 1.11.0...
===> Creating groups.
Using existing group 'unbound'.
===> Creating users
Using existing user 'unbound'.
[24/60] Extracting unbound-1.11.0: .......... done
[25/60] Upgrading syslog-ng327 from 3.27.1_1 to 3.27.1_2...
[25/60] Extracting syslog-ng327-3.27.1_2: .......... done
You may need to manually remove /usr/local/etc/syslog-ng.conf if it is no longer needed.
[26/60] Upgrading squid from 4.11_2 to 4.13...
===> Creating groups.
Using existing group 'squid'.
===> Creating users
Using existing user 'squid'.
===> Creating homedir(s)
===> Pre-installation configuration for squid-4.13
[26/60] Extracting squid-4.13: .......... done
You may need to manually remove /usr/local/etc/squid/squid.conf if it is no longer needed.
[27/60] Reinstalling rrdtool-1.7.2_4...
[27/60] Extracting rrdtool-1.7.2_4: .......... done
[28/60] Upgrading rate from 0.9_1 to 0.9_2...
[28/60] Extracting rate-0.9_2: ..... done
[29/60] Upgrading py37-sqlite3 from 3.7.8_7 to 3.7.9_7...
[29/60] Extracting py37-sqlite3-3.7.9_7: ........ done
[30/60] Upgrading php73-zlib from 7.3.20 to 7.3.22...
[30/60] Extracting php73-zlib-7.3.22: ....... done
[31/60] Upgrading php73-xml from 7.3.20 to 7.3.22...
[31/60] Extracting php73-xml-7.3.22: ........ done
[32/60] Upgrading php73-sqlite3 from 7.3.20 to 7.3.22...
[32/60] Extracting php73-sqlite3-7.3.22: ........ done
[33/60] Upgrading php73-sockets from 7.3.20 to 7.3.22...
[33/60] Extracting php73-sockets-7.3.22: .......... done
[34/60] Upgrading php73-simplexml from 7.3.20 to 7.3.22...
[34/60] Extracting php73-simplexml-7.3.22: ......... done
[35/60] Upgrading php73-session from 7.3.20 to 7.3.22...
[35/60] Extracting php73-session-7.3.22: .......... done
[36/60] Upgrading php73-openssl from 7.3.20 to 7.3.22...
[36/60] Extracting php73-openssl-7.3.22: ....... done
[37/60] Upgrading php73-ldap from 7.3.20 to 7.3.22...
[37/60] Extracting php73-ldap-7.3.22: ....... done
[38/60] Upgrading php73-gettext from 7.3.20 to 7.3.22...
[38/60] Extracting php73-gettext-7.3.22: ....... done
[39/60] Upgrading php73-filter from 7.3.20 to 7.3.22...
[39/60] Extracting php73-filter-7.3.22: ........ done
[40/60] Upgrading php73-dom from 7.3.20 to 7.3.22...
[40/60] Extracting php73-dom-7.3.22: .......... done
[41/60] Upgrading php73-curl from 7.3.20 to 7.3.22...
[41/60] Extracting php73-curl-7.3.22: ....... done
[42/60] Upgrading php73-ctype from 7.3.20 to 7.3.22...
[42/60] Extracting php73-ctype-7.3.22: ....... done
[43/60] Upgrading opnsense-update from 20.7 to 20.7.3...
[43/60] Extracting opnsense-update-20.7.3: .......... done
[44/60] Reinstalling ntp-4.2.8p15...
[44/60] Extracting ntp-4.2.8p15: .......... done
[45/60] Upgrading mpd5 from 5.8_10 to 5.9...
[45/60] Extracting mpd5-5.9: ......... done
[46/60] Upgrading isc-dhcp44-server from 4.4.2 to 4.4.2_1...
===> Creating groups.
Using existing group 'dhcpd'.
===> Creating users
Using existing user 'dhcpd'.
[46/60] Extracting isc-dhcp44-server-4.4.2_1: .......... done
[47/60] Upgrading isc-dhcp44-relay from 4.4.2 to 4.4.2_1...
[47/60] Extracting isc-dhcp44-relay-4.4.2_1: ....... done
[48/60] Upgrading frr7 from 7.4_1 to 7.4_2...
===> Creating groups.
Using existing group 'frr'.
Using existing group 'frrvty'.
===> Creating users
Using existing user 'frr'.
[48/60] Extracting frr7-7.4_2: .......... done
[49/60] Upgrading fping from 4.2 to 5.0_1...
[49/60] Extracting fping-5.0_1: ....... done
[50/60] Deinstalling zabbix4-agent-4.0.22...
[50/60] Deleting files for zabbix4-agent-4.0.22: .......... done
[51/60] Upgrading py37-six from 1.14.0 to 1.15.0...
[51/60] Extracting py37-six-1.15.0: .......... done
[52/60] Upgrading py37-cffi from 1.14.0_1 to 1.14.3...
[52/60] Extracting py37-cffi-1.14.3: .......... done
[53/60] Upgrading py37-urllib3 from 1.25.7,1 to 1.25.10,1...
[53/60] Extracting py37-urllib3-1.25.10,1: .......... done
[54/60] Upgrading nss from 3.55 to 3.57...
[54/60] Extracting nss-3.57: .......... done
[55/60] Upgrading libunwind from 20200331 to 20200331_1...
[55/60] Extracting libunwind-20200331_1: .......... done
[56/60] Upgrading zabbix4-proxy from 4.0.22 to 4.0.24...
===> Creating groups.
Using existing group 'zabbix'.
===> Creating users
Using existing user 'zabbix'.
[56/60] Extracting zabbix4-proxy-4.0.24: ......... done
[57/60] Upgrading os-frr from 1.14 to 1.17...
[57/60] Extracting os-frr-1.17: .......... done
Stopping configd...done
Starting configd.
Keep version OPNsense\Quagga\General (1.0.1)
Keep version OPNsense\Quagga\OSPF (1.0.4)
*** OPNsense\Quagga\OSPF6 Migration failed, check log for details
*** OPNsense\Quagga\RIP Migration failed, check log for details
*** OPNsense\Quagga\BGP Migration failed, check log for details
Reloading plugin configuration
Configuring system logging...done.
Reloading template OPNsense/Quagga: OK
```

**Comment by mimugmail** [comment]
Ah, now I updated the slave system (so the same as above) and error did not appeared and also no reinstall of pkg:


```
***GOT REQUEST TO UPGRADE: all***
Updating OPNsense repository catalogue...
OPNsense repository is up to date.
All repositories are up to date.
Updating OPNsense repository catalogue...
OPNsense repository is up to date.
All repositories are up to date.
Checking for upgrades (50 candidates): .......... done
Processing candidates (50 candidates): .......... done
The following 50 package(s) will be affected (of 0 checked):

Installed packages to be UPGRADED:
	ca_root_nss: 3.55 -> 3.57
	curl: 7.71.1 -> 7.72.0
	frr7: 7.4_1 -> 7.4_2
	gettext-runtime: 0.20.2 -> 0.21
	isc-dhcp44-relay: 4.4.2 -> 4.4.2_1
	isc-dhcp44-server: 4.4.2 -> 4.4.2_1
	json-c: 0.14 -> 0.15
	libffi: 3.3 -> 3.3_1
	libssh: 0.9.4 -> 0.9.5
	libunwind: 20200331 -> 20200331_1
	mpd5: 5.8_10 -> 5.9
	net-snmp: 5.7.3_20,1 -> 5.9,1
	nspr: 4.27 -> 4.29
	nss: 3.55 -> 3.57
	openldap-sasl-client: 2.4.50 -> 2.4.51
	opnsense: 20.7 -> 20.7.3
	opnsense-update: 20.7 -> 20.7.3
	os-dyndns: 1.22 -> 1.23
	os-frr: 1.14 -> 1.17
	perl5: 5.30.3 -> 5.32.0
	php73: 7.3.20 -> 7.3.22
	php73-ctype: 7.3.20 -> 7.3.22
	php73-curl: 7.3.20 -> 7.3.22
	php73-dom: 7.3.20 -> 7.3.22
	php73-filter: 7.3.20 -> 7.3.22
	php73-gettext: 7.3.20 -> 7.3.22
	php73-hash: 7.3.20 -> 7.3.22
	php73-json: 7.3.20 -> 7.3.22
	php73-ldap: 7.3.20 -> 7.3.22
	php73-openssl: 7.3.20 -> 7.3.22
	php73-pdo: 7.3.20 -> 7.3.22
	php73-session: 7.3.20 -> 7.3.22
	php73-simplexml: 7.3.20 -> 7.3.22
	php73-sockets: 7.3.20 -> 7.3.22
	php73-sqlite3: 7.3.20 -> 7.3.22
	php73-xml: 7.3.20 -> 7.3.22
	php73-zlib: 7.3.20 -> 7.3.22
	py37-cffi: 1.14.0_1 -> 1.14.3
	py37-six: 1.14.0 -> 1.15.0
	py37-sqlite3: 3.7.8_7 -> 3.7.9_7
	py37-urllib3: 1.25.7,1 -> 1.25.10,1
	python37: 3.7.8_1 -> 3.7.9
	rate: 0.9_1 -> 0.9_2
	sqlite3: 3.32.3_1,1 -> 3.33.0,1
	squid: 4.11_2 -> 4.13
	syslog-ng327: 3.27.1_1 -> 3.27.1_2
	unbound: 1.10.1 -> 1.11.0

Installed packages to be REINSTALLED:
	ntp-4.2.8p15 (direct dependency changed: perl5)
	rrdtool-1.7.2_4 (direct dependency changed: perl5)
	rtrlib-0.6.3 (needed shared library changed)

Number of packages to be upgraded: 47
Number of packages to be reinstalled: 3

The process will require 2 MiB more space.
58 MiB to be downloaded.
[1/50] Fetching unbound-1.11.0.txz: .......... done
[2/50] Fetching syslog-ng327-3.27.1_2.txz: .......... done
[3/50] Fetching squid-4.13.txz: .......... done
[4/50] Fetching sqlite3-3.33.0,1.txz: .......... done
[5/50] Fetching rtrlib-0.6.3.txz: ...... done
[6/50] Fetching rrdtool-1.7.2_4.txz: .......... done
[7/50] Fetching rate-0.9_2.txz: ....... done
[8/50] Fetching python37-3.7.9.txz: .......... done
[9/50] Fetching py37-urllib3-1.25.10,1.txz: .......... done
[10/50] Fetching py37-sqlite3-3.7.9_7.txz: .... done
[11/50] Fetching py37-six-1.15.0.txz: ... done
[12/50] Fetching py37-cffi-1.14.3.txz: .......... done
[13/50] Fetching php73-zlib-7.3.22.txz: ... done
[14/50] Fetching php73-xml-7.3.22.txz: ... done
[15/50] Fetching php73-sqlite3-7.3.22.txz: ... done
[16/50] Fetching php73-sockets-7.3.22.txz: ..... done
[17/50] Fetching php73-simplexml-7.3.22.txz: ... done
[18/50] Fetching php73-session-7.3.22.txz: ..... done
[19/50] Fetching php73-pdo-7.3.22.txz: ...... done
[20/50] Fetching php73-openssl-7.3.22.txz: ........ done
[21/50] Fetching php73-ldap-7.3.22.txz: .... done
[22/50] Fetching php73-json-7.3.22.txz: ... done
[23/50] Fetching php73-hash-7.3.22.txz: .......... done
[24/50] Fetching php73-gettext-7.3.22.txz: . done
[25/50] Fetching php73-filter-7.3.22.txz: ... done
[26/50] Fetching php73-dom-7.3.22.txz: ........ done
[27/50] Fetching php73-curl-7.3.22.txz: .... done
[28/50] Fetching php73-ctype-7.3.22.txz: . done
[29/50] Fetching php73-7.3.22.txz: .......... done
[30/50] Fetching perl5-5.32.0.txz: .......... done
[31/50] Fetching os-frr-1.17.txz: ....... done
[32/50] Fetching os-dyndns-1.23.txz: .... done
[33/50] Fetching opnsense-update-20.7.3.txz: ........ done
[34/50] Fetching opnsense-20.7.3.txz: .......... done
[35/50] Fetching openldap-sasl-client-2.4.51.txz: .......... done
[36/50] Fetching ntp-4.2.8p15.txz: .......... done
[37/50] Fetching nss-3.57.txz: .......... done
[38/50] Fetching nspr-4.29.txz: .......... done
[39/50] Fetching net-snmp-5.9,1.txz: .......... done
[40/50] Fetching mpd5-5.9.txz: .......... done
[41/50] Fetching libunwind-20200331_1.txz: .......... done
[42/50] Fetching libssh-0.9.5.txz: .......... done
[43/50] Fetching libffi-3.3_1.txz: ..... done
[44/50] Fetching json-c-0.15.txz: ......... done
[45/50] Fetching isc-dhcp44-server-4.4.2_1.txz: .......... done
[46/50] Fetching isc-dhcp44-relay-4.4.2_1.txz: .......... done
[47/50] Fetching gettext-runtime-0.21.txz: .......... done
[48/50] Fetching frr7-7.4_2.txz: .......... done
[49/50] Fetching curl-7.72.0.txz: .......... done
[50/50] Fetching ca_root_nss-3.57.txz: .......... done
Checking integrity... done (0 conflicting)
[1/50] Upgrading libffi from 3.3 to 3.3_1...
[1/50] Extracting libffi-3.3_1: .......... done
[2/50] Upgrading python37 from 3.7.8_1 to 3.7.9...
[2/50] Extracting python37-3.7.9: .......... done
[3/50] Upgrading py37-six from 1.14.0 to 1.15.0...
[3/50] Extracting py37-six-1.15.0: .......... done
[4/50] Upgrading py37-cffi from 1.14.0_1 to 1.14.3...
[4/50] Extracting py37-cffi-1.14.3: .......... done
[5/50] Upgrading sqlite3 from 3.32.3_1,1 to 3.33.0,1...
[5/50] Extracting sqlite3-3.33.0,1: .......... done
[6/50] Upgrading php73 from 7.3.20 to 7.3.22...
[6/50] Extracting php73-7.3.22: .......... done
[7/50] Upgrading perl5 from 5.30.3 to 5.32.0...
[7/50] Extracting perl5-5.32.0: .......... done
[8/50] Upgrading nspr from 4.27 to 4.29...
[8/50] Extracting nspr-4.29: .......... done
[9/50] Upgrading libssh from 0.9.4 to 0.9.5...
[9/50] Extracting libssh-0.9.5: .......... done
[10/50] Upgrading gettext-runtime from 0.20.2 to 0.21...
[10/50] Extracting gettext-runtime-0.21: .......... done
[11/50] Upgrading ca_root_nss from 3.55 to 3.57...
[11/50] Extracting ca_root_nss-3.57: ...... done
[12/50] Reinstalling rtrlib-0.6.3...
[12/50] Extracting rtrlib-0.6.3: .......... done
[13/50] Upgrading py37-urllib3 from 1.25.7,1 to 1.25.10,1...
[13/50] Extracting py37-urllib3-1.25.10,1: .......... done
[14/50] Upgrading php73-pdo from 7.3.20 to 7.3.22...
[14/50] Extracting php73-pdo-7.3.22: .......... done
[15/50] Upgrading php73-json from 7.3.20 to 7.3.22...
[15/50] Extracting php73-json-7.3.22: .......... done
[16/50] Upgrading php73-hash from 7.3.20 to 7.3.22...
[16/50] Extracting php73-hash-7.3.22: .......... done
[17/50] Upgrading openldap-sasl-client from 2.4.50 to 2.4.51...
[17/50] Extracting openldap-sasl-client-2.4.51: .......... done
[18/50] Upgrading nss from 3.55 to 3.57...
[18/50] Extracting nss-3.57: .......... done
[19/50] Upgrading net-snmp from 5.7.3_20,1 to 5.9,1...
[19/50] Extracting net-snmp-5.9,1: .......... done
[20/50] Upgrading libunwind from 20200331 to 20200331_1...
[20/50] Extracting libunwind-20200331_1: .......... done
[21/50] Upgrading json-c from 0.14 to 0.15...
[21/50] Extracting json-c-0.15: .......... done
[22/50] Upgrading curl from 7.71.1 to 7.72.0...
[22/50] Extracting curl-7.72.0: .......... done
[23/50] Upgrading unbound from 1.10.1 to 1.11.0...
===> Creating groups.
Using existing group 'unbound'.
===> Creating users
Using existing user 'unbound'.
[23/50] Extracting unbound-1.11.0: .......... done
[24/50] Upgrading syslog-ng327 from 3.27.1_1 to 3.27.1_2...
[24/50] Extracting syslog-ng327-3.27.1_2: .......... done
You may need to manually remove /usr/local/etc/syslog-ng.conf if it is no longer needed.
[25/50] Upgrading squid from 4.11_2 to 4.13...
===> Creating groups.
Using existing group 'squid'.
===> Creating users
Using existing user 'squid'.
===> Creating homedir(s)
===> Pre-installation configuration for squid-4.13
[25/50] Extracting squid-4.13: .......... done
You may need to manually remove /usr/local/etc/squid/squid.conf if it is no longer needed.
[26/50] Reinstalling rrdtool-1.7.2_4...
[26/50] Extracting rrdtool-1.7.2_4: .......... done
[27/50] Upgrading rate from 0.9_1 to 0.9_2...
[27/50] Extracting rate-0.9_2: ..... done
[28/50] Upgrading py37-sqlite3 from 3.7.8_7 to 3.7.9_7...
[28/50] Extracting py37-sqlite3-3.7.9_7: ........ done
[29/50] Upgrading php73-zlib from 7.3.20 to 7.3.22...
[29/50] Extracting php73-zlib-7.3.22: ....... done
[30/50] Upgrading php73-xml from 7.3.20 to 7.3.22...
[30/50] Extracting php73-xml-7.3.22: ........ done
[31/50] Upgrading php73-sqlite3 from 7.3.20 to 7.3.22...
[31/50] Extracting php73-sqlite3-7.3.22: ........ done
[32/50] Upgrading php73-sockets from 7.3.20 to 7.3.22...
[32/50] Extracting php73-sockets-7.3.22: .......... done
[33/50] Upgrading php73-simplexml from 7.3.20 to 7.3.22...
[33/50] Extracting php73-simplexml-7.3.22: ......... done
[34/50] Upgrading php73-session from 7.3.20 to 7.3.22...
[34/50] Extracting php73-session-7.3.22: .......... done
[35/50] Upgrading php73-openssl from 7.3.20 to 7.3.22...
[35/50] Extracting php73-openssl-7.3.22: ....... done
[36/50] Upgrading php73-ldap from 7.3.20 to 7.3.22...
[36/50] Extracting php73-ldap-7.3.22: ....... done
[37/50] Upgrading php73-gettext from 7.3.20 to 7.3.22...
[37/50] Extracting php73-gettext-7.3.22: ....... done
[38/50] Upgrading php73-filter from 7.3.20 to 7.3.22...
[38/50] Extracting php73-filter-7.3.22: ........ done
[39/50] Upgrading php73-dom from 7.3.20 to 7.3.22...
[39/50] Extracting php73-dom-7.3.22: .......... done
[40/50] Upgrading php73-curl from 7.3.20 to 7.3.22...
[40/50] Extracting php73-curl-7.3.22: ....... done
[41/50] Upgrading php73-ctype from 7.3.20 to 7.3.22...
[41/50] Extracting php73-ctype-7.3.22: ....... done
[42/50] Upgrading opnsense-update from 20.7 to 20.7.3...
[42/50] Extracting opnsense-update-20.7.3: .......... done
[43/50] Reinstalling ntp-4.2.8p15...
[43/50] Extracting ntp-4.2.8p15: .......... done
[44/50] Upgrading mpd5 from 5.8_10 to 5.9...
[44/50] Extracting mpd5-5.9: ......... done
[45/50] Upgrading isc-dhcp44-server from 4.4.2 to 4.4.2_1...
===> Creating groups.
Using existing group 'dhcpd'.
===> Creating users
Using existing user 'dhcpd'.
[45/50] Extracting isc-dhcp44-server-4.4.2_1: .......... done
[46/50] Upgrading isc-dhcp44-relay from 4.4.2 to 4.4.2_1...
[46/50] Extracting isc-dhcp44-relay-4.4.2_1: ....... done
[47/50] Upgrading frr7 from 7.4_1 to 7.4_2...
===> Creating groups.
Using existing group 'frr'.
Using existing group 'frrvty'.
===> Creating users
Using existing user 'frr'.
[47/50] Extracting frr7-7.4_2: .......... done
[48/50] Upgrading os-frr from 1.14 to 1.17...
[48/50] Extracting os-frr-1.17: .......... done
Stopping configd...done
Starting configd.
Keep version OPNsense\Quagga\General (1.0.1)
Keep version OPNsense\Quagga\OSPF (1.0.4)
*** OPNsense\Quagga\OSPF6 Migration failed, check log for details
*** OPNsense\Quagga\RIP Migration failed, check log for details
*** OPNsense\Quagga\BGP Migration failed, check log for details
Reloading plugin configuration
Configuring system logging...done.
Reloading template OPNsense/Quagga: OK
[49/50] Upgrading os-dyndns from 1.22 to 1.23...
[49/50] Extracting os-dyndns-1.23: .......... done
Stopping configd...done
Starting configd.
Reloading plugin configuration
Configuring system logging...done.
[50/50] Upgrading opnsense from 20.7 to 20.7.3...
[50/50] Extracting opnsense-20.7.3: .......... done
Stopping configd...done
Resetting root shell
Updating /etc/shells
Unhooking from /etc/rc
Unhooking from /etc/rc.shutdown
Updating /etc/shells
Registering root shell
Hooking into /etc/rc
Hooking into /etc/rc.shutdown
Starting configd.
Keep version OPNsense\CaptivePortal\CaptivePortal (1.0.0)
Keep version OPNsense\Cron\Cron (1.0.1)
Keep version OPNsense\TrafficShaper\TrafficShaper (1.0.3)
Keep version OPNsense\IDS\IDS (1.0.5)
Keep version OPNsense\Proxy\Proxy (1.0.3)
Keep version OPNsense\Diagnostics\Netflow (1.0.1)
Keep version OPNsense\Routes\Route (1.0.0)
Keep version OPNsense\Backup\NextcloudSettings (1.0.0)
Keep version OPNsense\Firewall\Alias (1.0.0)
Keep version OPNsense\Monit\Monit (1.0.8)
Keep version OPNsense\OpenVPN\Export (0.0.1)
Keep version OPNsense\Syslog\Syslog (1.0.0)
Keep version OPNsense\Quagga\General (1.0.1)
Keep version OPNsense\Quagga\OSPF (1.0.4)
*** OPNsense\Quagga\OSPF6 Migration failed, check log for details
*** OPNsense\Quagga\RIP Migration failed, check log for details
*** OPNsense\Quagga\BGP Migration failed, check log for details
Keep version OPNsense\IPsec\IPsec (0.0.0)
Keep version OPNsense\Interfaces\VxLan (1.0.1)
Keep version OPNsense\Interfaces\Loopback (1.0.0)
Keep version OPNsense\Unboundplus\Miscellaneous (0.0.2)
Keep version OPNsense\Unboundplus\Dnsbl (0.0.1)
Writing firmware setting...done.
Writing trust files...done.
Configuring login behaviour...done.
Configuring system logging...done.
=====
Message from py37-urllib3-1.25.10,1:

--
Since version 1.25 HTTPS connections are now verified by default which is done
via "cert_reqs = 'CERT_REQUIRED'".  While certificate verification can be
disabled via "cert_reqs = 'CERT_NONE'", it's highly recommended to leave it on.

Various consumers of net/py-urllib3 already have implemented routines that
either explicitly enable or disable HTTPS certificate verification (e.g. via
configuration settings, CLI arguments, etc.).

Yet it may happen that there are still some consumers which don't explicitly
enable/disable certificate verification for HTTPS connections which could then
lead to errors (as is often the case with self-signed certificates).

In case of an error one should try first to temporarily disable certificate
verification of the problematic urllib3 consumer to see if that approach will
remedy the issue.
=====
Message from opnsense-20.7.3:

--
The lion sleeps tonight
Checking integrity... done (0 conflicting)
Nothing to do.
The following package files will be deleted:
	/var/cache/pkg/pkg-1.15.6-5457c2d69b.txz
	/var/cache/pkg/unbound-1.11.0.txz
	/var/cache/pkg/pkg-1.15.6.txz
	/var/cache/pkg/unbound-1.11.0~ab09f69c09.txz
	/var/cache/pkg/syslog-ng327-3.27.1_2~1ad11b98f9.txz
	/var/cache/pkg/squid-4.13~21f2feca67.txz
	/var/cache/pkg/syslog-ng327-3.27.1_2.txz
	/var/cache/pkg/rtrlib-0.6.3.txz
	/var/cache/pkg/squid-4.13.txz
	/var/cache/pkg/sqlite3-3.33.0,1~db80df7f62.txz
	/var/cache/pkg/rtrlib-0.6.3~0d9e779169.txz
	/var/cache/pkg/sqlite3-3.33.0,1.txz
	/var/cache/pkg/rrdtool-1.7.2_4~74cbbb5144.txz
	/var/cache/pkg/rate-0.9_2~be0f156945.txz
	/var/cache/pkg/rrdtool-1.7.2_4.txz
	/var/cache/pkg/python37-3.7.9.txz
	/var/cache/pkg/rate-0.9_2.txz
	/var/cache/pkg/python37-3.7.9~0349942bbf.txz
	/var/cache/pkg/py37-urllib3-1.25.10,1~375e46b391.txz
	/var/cache/pkg/py37-six-1.15.0~935d1cec33.txz
	/var/cache/pkg/py37-urllib3-1.25.10,1.txz
	/var/cache/pkg/py37-sqlite3-3.7.9_7~897bcd2ba6.txz
	/var/cache/pkg/py37-sqlite3-3.7.9_7.txz
	/var/cache/pkg/py37-cffi-1.14.3.txz
	/var/cache/pkg/py37-six-1.15.0.txz
	/var/cache/pkg/py37-cffi-1.14.3~07babc08bf.txz
	/var/cache/pkg/php73-zlib-7.3.22~e3f8120e05.txz
	/var/cache/pkg/php73-xml-7.3.22~28041d9d2a.txz
	/var/cache/pkg/php73-zlib-7.3.22.txz
	/var/cache/pkg/php73-sqlite3-7.3.22.txz
	/var/cache/pkg/php73-xml-7.3.22.txz
	/var/cache/pkg/php73-sqlite3-7.3.22~cb38bfeb75.txz
	/var/cache/pkg/php73-sockets-7.3.22~cf97767d45.txz
	/var/cache/pkg/php73-pdo-7.3.22~d885ed2567.txz
	/var/cache/pkg/php73-sockets-7.3.22.txz
	/var/cache/pkg/php73-simplexml-7.3.22~8090f07f1e.txz
	/var/cache/pkg/php73-simplexml-7.3.22.txz
	/var/cache/pkg/php73-session-7.3.22~13f2115548.txz
	/var/cache/pkg/php73-session-7.3.22.txz
	/var/cache/pkg/php73-pdo-7.3.22.txz
	/var/cache/pkg/php73-openssl-7.3.22~23222d476a.txz
	/var/cache/pkg/php73-dom-7.3.22~1673443386.txz
	/var/cache/pkg/php73-openssl-7.3.22.txz
	/var/cache/pkg/php73-ldap-7.3.22~bc625fd252.txz
	/var/cache/pkg/php73-ldap-7.3.22.txz
	/var/cache/pkg/php73-json-7.3.22~85776048d5.txz
	/var/cache/pkg/php73-json-7.3.22.txz
	/var/cache/pkg/php73-hash-7.3.22~54181f600c.txz
	/var/cache/pkg/php73-hash-7.3.22.txz
	/var/cache/pkg/php73-gettext-7.3.22~e9f5f6bdb2.txz
	/var/cache/pkg/php73-gettext-7.3.22.txz
	/var/cache/pkg/php73-filter-7.3.22~ebe1ce9271.txz
	/var/cache/pkg/php73-filter-7.3.22.txz
	/var/cache/pkg/php73-curl-7.3.22~36dc4ffb6b.txz
	/var/cache/pkg/php73-dom-7.3.22.txz
	/var/cache/pkg/php73-curl-7.3.22.txz
	/var/cache/pkg/php73-ctype-7.3.22~b9f77945fa.txz
	/var/cache/pkg/php73-7.3.22~6820b9e298.txz
	/var/cache/pkg/php73-ctype-7.3.22.txz
	/var/cache/pkg/php73-7.3.22.txz
	/var/cache/pkg/perl5-5.32.0~4342008c40.txz
	/var/cache/pkg/perl5-5.32.0.txz
	/var/cache/pkg/os-frr-1.17~642cc94a78.txz
	/var/cache/pkg/os-frr-1.17.txz
	/var/cache/pkg/os-dyndns-1.23~620a60b0bf.txz
	/var/cache/pkg/os-dyndns-1.23.txz
	/var/cache/pkg/opnsense-update-20.7.3~5c594cd5e9.txz
	/var/cache/pkg/opnsense-20.7.3~974da3e0a0.txz
	/var/cache/pkg/opnsense-20.7.3.txz
	/var/cache/pkg/opnsense-update-20.7.3.txz
	/var/cache/pkg/openldap-sasl-client-2.4.51~bc9d6fb838.txz
	/var/cache/pkg/ntp-4.2.8p15~670cafefb1.txz
	/var/cache/pkg/openldap-sasl-client-2.4.51.txz
	/var/cache/pkg/nss-3.57~af91bd1182.txz
	/var/cache/pkg/ntp-4.2.8p15.txz
	/var/cache/pkg/nss-3.57.txz
	/var/cache/pkg/nspr-4.29~eda6aa6ac3.txz
	/var/cache/pkg/nspr-4.29.txz
	/var/cache/pkg/net-snmp-5.9,1~d83aa76108.txz
	/var/cache/pkg/mpd5-5.9~760378618a.txz
	/var/cache/pkg/net-snmp-5.9,1.txz
	/var/cache/pkg/mpd5-5.9.txz
	/var/cache/pkg/libunwind-20200331_1~c8e453ddc3.txz
	/var/cache/pkg/libunwind-20200331_1.txz
	/var/cache/pkg/libssh-0.9.5~c371d9b650.txz
	/var/cache/pkg/libssh-0.9.5.txz
	/var/cache/pkg/libffi-3.3_1~49b953d00b.txz
	/var/cache/pkg/libffi-3.3_1.txz
	/var/cache/pkg/json-c-0.15~0283a80580.txz
	/var/cache/pkg/json-c-0.15.txz
	/var/cache/pkg/isc-dhcp44-server-4.4.2_1~1d10fa7f54.txz
	/var/cache/pkg/isc-dhcp44-server-4.4.2_1.txz
	/var/cache/pkg/isc-dhcp44-relay-4.4.2_1~adade75c88.txz
	/var/cache/pkg/gettext-runtime-0.21~28fefd2893.txz
	/var/cache/pkg/isc-dhcp44-relay-4.4.2_1.txz
	/var/cache/pkg/frr7-7.4_2~470a9fc2c7.txz
	/var/cache/pkg/gettext-runtime-0.21.txz
	/var/cache/pkg/frr7-7.4_2.txz
	/var/cache/pkg/curl-7.72.0~e401be138c.txz
	/var/cache/pkg/curl-7.72.0.txz
	/var/cache/pkg/ca_root_nss-3.57~df7fc72e32.txz
	/var/cache/pkg/ca_root_nss-3.57.txz
The cleanup will free 62 MiB
Deleting files: .......... done
All done
Starting web GUI...done.
Generating RRD graphs...done.
Fetching base-20.7.3-amd64.txz: ...
```

**Comment by fichtner** [comment]
> no reinstall of pkg

huh, pkg as per its own requirement is always separately upgraded

**Comment by mimugmail** [comment]
Yes, but as you can see in the first comment (where update stucks) it tries to reinstall pkg for whatevery reason:

```
Installed packages to be REINSTALLED:
	ntp-4.2.8p15 (direct dependency changed: perl5)
	pkg-1.15.6
	rrdtool-1.7.2_4 (direct dependency changed: perl5)
	rtrlib-0.6.3 (needed shared library changed)

Number of packages to be installed: 2
Number of packages to be upgraded: 52
Number of packages to be reinstalled: 4

The process will require 5 MiB more space.
4 MiB to be downloaded.
[1/1] Fetching pkg-1.15.6.txz: .......... done
[1/60] Upgrading libffi from 3.3 to 3.3_1...
[1/60] Extracting libffi-3.3_1: .......... done
```

**Comment by fichtner** [comment]
my guess is net-smtp which has pkg as a dependency...

**Comment by mimugmail** [comment]
No, I also had this on fresh installed 20.7 img's without config or plugins. just WAN DHCP and go for 20.7.3 ...

**Comment by fichtner** [comment]
I am unsure what the issue is beyond the pkg stall bug not being fixed properly. Looking at the inner workings of the package dependency resolver is not overly helpful from past experiences.

**Comment by fichtner** [comment]
For reference the bug seemed to be this, got fixed but not entirely as it would seem https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=245462

**Comment by fichtner** [comment]
Just FYI, the best way to deal with this for now is:

    # killall syslog-ng

and let pkg continue. I'm not sure if this an issue with syslog-ng / shared library randomization which was fixed in 20.7.3 so updates from 20.7 - 20.7.2 exhibit this issue (I just saw it with a fresh 20.7).

**Comment by fichtner** [comment]
Actually I just had a different issue that looked similar but wasn't: pkg gave no indication that it wasn't done but the GUI firmware was still thinking the update didn't finish most likely because the ***DONE*** was not present in the output yet.

**Comment by fichtner** [comment]
Something else I did there changing syslog-ng.pid to the wrong location.

**Comment by mimugmail** [comment]
I can test on a couple of devices if you push everything to snapshots mirror

**Comment by mimugmail** [comment]
I updated couple of device (all were on 20.7.3) and it happened one, your workaround helped without the need for a second reboot.

**Comment by AdSchellevis** [comment]
@fichtner https://github.com/opnsense/core/commit/186a8815b80b0bbc5df7462f4cb3e3086bb5957c this is what I meant yesterday. if we detach the reconfiguration to configd pkg likely won't keep waiting on newly spawned processes.

**Comment by fichtner** [comment]
Ok that could work 👍

**Comment by fichtner** [comment]
I'm marking this as upstream now since we did what we can from our end.

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=250888 -> https://github.com/freebsd/pkg/commit/d40dbc609a

A bit too intransparent for my taste with the refactoring hiding the actual bug fix but we shall see when pkg 1.16 arrives.

**Comment by fichtner** [comment]
We decided to go ahead with a backport https://github.com/opnsense/ports/commit/f671decd08d to be used in 20.7.5.
